pkg/registry/core/pod/storage/eviction.go

```diff
@@ -138,43 +138,44 @@ func (r *EvictionREST) Create(ctx context.Context, name string, obj runtime.Obje
 		return &metav1.Status{
 			Status: metav1.StatusSuccess}, nil
 	}
-	var rtStatus *metav1.Status
 	var pdbName string
 	err = func() error {
 		pdbs, err := r.getPodDisruptionBudgets(ctx, pod)
 		if err != nil {
 			return err
 		}
 
-		if len(pdbs) > 1 {
-			rtStatus = &metav1.Status{
-				Status:  metav1.StatusFailure,
-				Message: "This pod has more than one PodDisruptionBudget, which the eviction subresource does not support.",
-				Code:    500,
-			}
-			return nil
-		}
 		if len(pdbs) == 0 {
 			return nil
 		}
 
-		pdb := &pdbs[0]
-		pdbName = pdb.Name
+		var pdb *policyv1beta1.PodDisruptionBudget
 		refresh := false
 		err = retry.RetryOnConflict(EvictionsRetry, func() error {
-			if refresh {
-				pdb, err = r.podDisruptionBudgetClient.PodDisruptionBudgets(pod.Namespace).Get(pdbName, metav1.GetOptions{})
-				if err != nil {
-					return err
+			for _, withDryRun := range []bool{true, false} {
+				for i := range pdbs {
+					pdbName = pdbs[i].Name
+					if refresh {
+						pdb, err = r.podDisruptionBudgetClient.PodDisruptionBudgets(pod.Namespace).Get(pdbName, metav1.GetOptions{})
+						if err != nil {
+							return err
+						}
+					} else {
+						pdb = &pdbs[i]
+					}
+					// Try to verify-and-decrement
+
+					// If it was false already, or if it becomes false during the course of our retries,
+					// raise an error marked as a 429.
+					if err = r.checkAndDecrement(pod.Namespace, pod.Name, *pdb, withDryRun); err != nil {
+						refresh = true
```

Comment:
I don't think we can set refresh to true here. If the first PDB is successfully decremented, we might still need to refresh the remaining PDBs. If we have two PDBs and we get a conflict on the last one, we will retry both of them, set the refresh variable to true after the first one and then retry the second PDB with the old pdb, which will fail with conflict again.
I think it might be easier if we retry on a per-pdb basis instead of retrying all pdbs if we get a conflict on one of them. And once we have completed the first dry-run and start updating the PDBs, it shouldn't be necessary to do any more dry-runs.